src/structures/GuildChannel.js

```diff
@@ -37,7 +36,11 @@ class GuildChannel extends Channel {
     this.guild = guild;
 
     this.parentID = this.parentID ?? null;
-    this.permissionOverwrites = this.permissionOverwrites ?? new Collection();
+    /**
+     * A manager of permission overwrites that belong to this channel
+     * @type {PermissionOverwriteManager}
+     */
+    this.permissionOverwrites = new PermissionOverwriteManager(this);
```

Comment:
This line will be executed _after_ the base class called `_patch`, causing said method to throw since the permission overwrite manager isn't set yet.

If you rebase this PR onto master, you can pass a third parameter (`immediatePatch`) to the super call causing the base class not to call `_patch`, allowing you to call it here yourself after the manager has been set.